src/geohash_helper.c

```diff
@@ -254,8 +258,11 @@ int geohashGetDistanceIfInRadiusWGS84(double x1, double y1, double x2,
 int geohashGetDistanceIfInRectangle(double width_m, double height_m, double x1, double y1,
                                     double x2, double y2, double *distance) {
     double lon_distance = geohashGetDistance(x2, y2, x1, y2);
+    if (lon_distance > width_m/2) {
+        return 0;
+    }
     double lat_distance = geohashGetDistance(x2, y2, x2, y1);
-    if (lon_distance > width_m/2 || lat_distance > height_m/2) {
+    if (lat_distance > height_m/2) {
         return 0;
     }
```

Comment:
is the order here right? i'm having hard time to confirm.
we wanna make the expensive one second, and the cheap one first.
in the first one, we're using the same `y` (lat), so `u` is 0.
seems like we better swap them around.
let's also add a comment.

Reply:
@oranagra you're absolutely right!
Swapped the code check and now latency dropped from 79ms to 73ms.
Added comment to explain it for future reference